File: backend/src/main/java/com/sellerops/collect/runtime/RowCountBucket.java
package com.sellerops.collect.runtime;

/**
 * Coarse count bucket for sanitized output. Exact row counts are an internal/DB concern;
 * anything logged or exposed uses a bucket so a precise volume never leaks.
 */
public enum RowCountBucket {
    ZERO,
    ONE,
    FEW,            // 2-9
    TENS,           // 10-99
    HUNDREDS,       // 100-999
    THOUSANDS_PLUS; // 1000+

    public static RowCountBucket of(int n) {
        if (n <= 0) {
            return ZERO;
        }
        if (n == 1) {
            return ONE;
        }
        if (n < 10) {
            return FEW;
        }
        if (n < 100) {
            return TENS;
        }
        if (n < 1000) {
            return HUNDREDS;
        }
        return THOUSANDS_PLUS;
    }
}

File: collector/src/row-count-bucket.ts
/**
 * Coarse row-count bucketing — the shared vocabulary for reporting "how many rows" without ever
 * emitting the exact number (`CLAUDE.md` §4.3).
 *
 * A zero-import pure leaf (§5) so any layer may depend on it: `upload.ts` needs it for its logs and
 * cannot reach `naver/review-upload-diagnostic.ts` (which imports `../upload`, so that edge would be
 * a cycle). `review-upload-diagnostic.ts` re-exports `rowCountBucket` under its established
 * `countBucket` name.
 *
 * `esm/esm-review-schema-shape.ts` still carries an identical private copy; folding the esm family in
 * is a separate slice.
 */

/** Coarse row-count bucket — never the exact count. */
export type RowCountBucket = "zero" | "one" | "few" | "tens" | "hundreds" | "thousands_plus";

/** Pure: coarse row-count bucket (never the exact count). */
export function rowCountBucket(n: number): RowCountBucket {
  if (!Number.isFinite(n) || n <= 0) return "zero";
  if (n === 1) return "one";
  if (n <= 9) return "few";
  if (n <= 99) return "tens";
  if (n <= 999) return "hundreds";
  return "thousands_plus";
}
